rasa/core/channels/slack.py

```diff
@@ -399,7 +457,16 @@ async def health(_: Request) -> HTTPResponse:
         @slack_webhook.route("/webhook", methods=["GET", "POST"])
         async def webhook(request: Request) -> HTTPResponse:
             content_type = request.headers.get("content-type")
-            # Slack API sends either a JSON-encoded or a URL-encoded body depending on the content
+
+            if not self.is_request_from_slack_authentic(request):
+                return response.text(
+                    "Message is not properly signed with a valid "
+                    "X-Slack-Signature header",
+                    status=400,
```

Comment:
this seems to fit better, what do you think?
```suggestion
                    status=HTTPStatus.UNAUTHORIZED,
```

Reply:
I don't think it is common to use that code for this reason as the response would need to contain an authenticate header.

Reply:
Mhm, but this way we can tell the client that they have to authenticate, right?

> Although the HTTP standard specifies "unauthorized", semantically this response means "unauthenticated". That is, the client must authenticate itself to get the requested response.

Reply:
this isn't a standard authentication though so there is no way to send a proper authenticate header back to the requester. anyways, this is a response to slack, so doesn't really matter and a 400 as a general user error should be fine